src/pac_array.cc

```diff
@@ -383,11 +383,9 @@ void ArrayType::GenCleanUpCode(Output* out_cc, Env* env)
 		out_cc->inc_indent();
 		out_cc->println("{");
 
-		out_cc->println("for ( int i = 0; i < (int) %s->size(); ++i )", env->RValue(value_var()));
-		out_cc->inc_indent();
+		out_cc->println("for ( auto* %s : *%s )", env->LValue(elem_var()),
+		                env->RValue(value_var()));
```

Comment:
Are you dropping an indentation here that still gets unindented below?

Reply:
Looks like I did, yeah. I'll stick it back in there.